### ISSUE TITLE
Add "Save Items" Feature

### ISSUE THREAD

**Issue by Traveller08** [Traveller08/wecode#24]
**Description:**

We would like to introduce a "Save Items" feature to enhance user experience. Users should be able to save posts, questions, and blogs for future reference. The "Save" option will be accessible by clicking the three-dot button on these items.

**Feature Requirements:**
- Add a "Save" option to posts, questions, and blogs accessible through a three-dot menu.
- Implement a system to save and organize these items for each user.
- "Saved Items" should be accessed by clicking the "saved items"  from the user's profile dropdown.
- Allow users to view and manage their saved items.

**Suggested Approach:**
- Design the user interface for the "Save" option and the "Saved Items" section.
- Implement the frontend components for saving items and managing saved items.
- Develop the backend logic to store and retrieve saved items for each user.
- Ensure a seamless and user-friendly experience.

**Additional Information:**
- Any specific technologies or libraries to consider.
- Compatibility considerations with the existing codebase.
- For each type of saved item (posts, questions, blogs), you create a separate table (e.g., Saved Posts Table, Saved Questions Table, Saved Blogs Table) to store the associations between users and their saved items. Each of these tables contains primary key saveid, userid , foreign keys to reference the specific item type (e.g., post_id, question_id, blog_id) ,  and a timestamp to record when the item was saved.

**Saved Posts Table:**

- save_id (Primary Key)
- user_id (Foreign Key referencing the User Table)
- post_id (Foreign Key referencing the Post Table)
- timestamp (Timestamp of when the post was saved)

Saved Questions Table:

- save_id (Primary Key)
- user_id (Foreign Key referencing the User Table)
- question_id (Foreign Key referencing the Question Table)
- timestamp (Timestamp of when the question was saved)

Saved Blogs Table:

- save_id (Primary Key)
- user_id (Foreign Key referencing the User Table)
- blog_id (Foreign Key referencing the Blog Table)
- timestamp (Timestamp of when the blog was saved)

**Acceptance Criteria:**
- After the implementation, users should be able to save and manage posts, questions, and blogs through the three-dot menu, with a "Saved Items" section accessible from the user's profile dropdown.

**Comment by mahek15** [comment]
Hey @Traveller08, I'd like to work on this feature.

**Comment by Traveller08** [comment]
Hi @mahek15 You can work on this feature.

**Comment by mahek15** [comment]
Hey, Can you provide me more information on exactly where you want this to be implemented?

**Comment by Traveller08** [comment]
See if you will create some posts after signing in, you will see the three dots on the posts, after which you will see the save option, so add your frontend code here. Same for the questions as well.

**Comment by mahek15** [comment]
Okay, Thankyou.

**Comment by mahek15** [comment]
Hey @Traveller08 , I have made a table for saved posts and can also display it in the saved items section. 

Where is the questions part that needs such a table too?

**Comment by Traveller08** [comment]
Homepage -> ask something
All the questions asked by the user will appear here. Like posts user should also be able to save these questions as well.

**Comment by Traveller08** [comment]
yes, either you can add a save button on the blogs and questions directly or can can add a three dot button that will open a dropdown and add save option there or any better solution.
